tools/check-development-environment.sh

```diff
@@ -354,7 +354,7 @@ else
 	version_range 'Git' "$BIN" '' "$VER" "0" "$GITVER" "9999999"
 
 	checking 'If this is a git checkout'
-	if [[ ! -d .git ]]; then
+	if [[ ! -d $(git rev-parse --git-dir) ]]; then
```

Comment:
In the failure case, this will result in some error output disrupting the output of the script.
```
* Usable version of git                    ... ok (version 2.33.0)
* If this is a git checkout                ... fatal: not a git repository (or any parent up to mount point /usr)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
no
  See https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#clone-the-repository
```

The test would also fail if Jetpack is copied into some other git repo instead of being checked out as a submodule. We might instead do something like
```bash
    checking 'If this is a git checkout'
    OK=false
    if [[ -d .git && "$(git rev-parse --git-dir 2>/dev/null)" == '.git' ]]; then
        success 'yes'
        OK=true
    elif [[ -f .git && -d "$(git rev-parse --git-dir 2>/dev/null)" ]]; then
        success 'yes (as a submodule)'
        OK=true
    elif [[ -d "$(git rev-parse --git-dir 2>/dev/null)" ]]; then
        failure 'unknown' 'clone-the-repository' "It seems to be in a git repo, but it's not clearly a Jetpack monorepo checkout."
    else
        failure "no" 'clone-the-repository'
    fi  
    if $OK; then
        checking 'If the repo is checked out using ssh'
        ...
```